test/unit/slave/test_cluster_slave.py

```diff
@@ -237,6 +244,34 @@ def test_execute_subjob_passes_base_executor_index_to_executor(self):
 
         executor.execute_subjob.assert_called_with(1, 2, [], 12)
 
+    @genty_dataset(
+        responsive_master=(True,1,),
+        unresponsive_master=(False,1,),
+        unresponsive_master_retry=(False,2,),
+    )
+    def test_heartbeat_sends_post_to_master(self, is_master_responsive, heartbeat_failure_threshold):
+        expected_slave_heartbeat_url = 'http://{}/v1/slave/1/heartbeat'.format(self._FAKE_MASTER_URL)
+        Configuration['heartbeat_failure_threshold'] = heartbeat_failure_threshold
+
+        slave = self._create_cluster_slave()
+        slave.connect_to_master(self._FAKE_MASTER_URL)
+        slave.kill = Mock()
```

Comment:
Same comment as above. Also, I'd rather us not mock out methods on the class under test. Assert on the behavior of the method, not the method itself being called.